internal/engine/build_and_deployer_test.go

```diff
@@ -202,21 +187,29 @@ func newBDFixture(t *testing.T, env k8s.Env, withPod bool) *bdFixture {
 			},
 		},
 	}
+	ctx := output.CtxForTest()
 	k8s := &FakeK8sClient{}
-	k8s.podWithImageExists = withPod
-	bd, err := provideBuildAndDeployer(output.CtxForTest(), docker, k8s, dir, env, synclet.NewFakeSyncletClient(), shouldFallBack)
+	bd, err := provideBuildAndDeployer(ctx, docker, k8s, dir, env, synclet.NewFakeSyncletClient(), shouldFallBack)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	return &bdFixture{
 		TempDirFixture: f,
+		ctx:            ctx,
 		docker:         docker,
 		k8s:            k8s,
 		bd:             bd,
 	}
 }
 
+// Ensure that the BuildAndDeployer has container information attached for the given manifest.
+func (f *bdFixture) withContainerForManifest(manifest model.Manifest, build BuildResult) *bdFixture {
+	f.k8s.podWithImageExists = true
```

Comment:
fyi, i deleted podWithImageExists in the other pr...there didn't seem to be a good reason for it to exist, all the tests pass without it.